include/includes_file.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef INCLUDESFILE_HPP
```

Comment:
It is generally not a good idea to have a single file with all includes. Instead, in every file (both hpp and cpp) you should all and only the headers that are actually used within that file. This usually reduces compilation times and prunes the dependency graph that makefiles use to detremine what source files to recompile (which further reduces the compilation times).